src/compile/data.ts

```diff
@@ -564,7 +667,16 @@ export namespace summary {
     return [];
   }
 
-  export function assemble(component: DataComponent): VgData[] {
+  export function parseLayer(model: LayerModel): SummaryComponent[] {
+    // we create new data sources for each summary in a layer
+    return [];
```

Comment:
The right thing to do is to move all summary from its children.  (remove from children) and merge summary with the same group by fields and do the rename in here.